### PR TITLE
Refactor out server

### PR DESCRIPTION
**Pull request by erh**
This is not 100% complete, but its a good staging point and don't want to keep diverging from the main line.

### CODE REVIEW

File: wire_message.go [hunk 1]
```diff
@@ -138,13 +138,13 @@ func parseBodySection(buf []byte, loc *int) (MessageMessageSection, error) {
 func parseDocumentSequenceSection(buf []byte, loc *int) (MessageMessageSection, error) {
 	dss := &DocumentSequenceSection{}
 
-	if len(buf[*loc:]) < 4 {
+	expectedSizeRemaining := readInt32(buf[*loc:])
```
Comment (tolsen) [inline wire_message.go]: This file seems to be missing changes I made to my op-msg branch after you branched from it.  Possibly a bad merge from master?

Comment (erh) [reply wire_message.go]: Is that the only one or is there a diff or are there more that are missing?
I thought I had everything...

**Comment by erh** [comment]
I pulled in wire_message.go from master.
Anything else look off?

**Comment by tolsen** [comment]
ugghh.. this probably happened because I deleted the op-msg branch after merging into master.  Sorry.  It looks like we have all the changes now.

**Pull request merged**